source/postprocess/particles.cc

```diff
@@ -31,21 +32,119 @@ namespace aspect
 {
   namespace Postprocess
   {
+#if DEAL_II_VERSION_GTE(9,0,0)
+    namespace internal
+    {
+      template<int dim>
+      void
+      ParticleOutput<dim>::build_patches(const std::multimap<aspect::Particle::types::LevelInd, aspect::Particle::Particle<dim> > &particles,
+                                         const aspect::Particle::Property::ParticlePropertyInformation &property_information)
+      {
+        // First store the names of the data fields
+        dataset_names.reserve(property_information.n_components()+1);
+        dataset_names.push_back("id");
+
+        for (unsigned int field_index = 0; field_index < property_information.n_fields(); ++field_index)
+          {
+            const unsigned n_components = property_information.get_components_by_field_index(field_index);
+            const std::string field_name = property_information.get_field_name_by_index(field_index);
+            // If it is a 1D element, or a vector, print just the name, otherwise use []
+            if ((n_components == 1) || (n_components == dim))
+              for (unsigned int component_index=0; component_index<n_components; ++component_index)
+                dataset_names.push_back(field_name);
+            else
+              for (unsigned int component_index=0; component_index<n_components; ++component_index)
+                dataset_names.push_back(field_name + "_" + Utilities::to_string(component_index));
+          }
+
+
+        // Second store which of these data fields are vectors
+
+        for (unsigned int field_index = 0; field_index < property_information.n_fields(); ++field_index)
+          {
+            const unsigned n_components = property_information.get_components_by_field_index(field_index);
+
+            // If the property has dim components, we treat it as vector
+            if (n_components == dim)
+              {
+                const unsigned int field_position = property_information.get_position_by_field_index(field_index);
+                const std::string field_name = property_information.get_field_name_by_index(field_index);
+                vector_datasets.push_back(std::make_tuple(field_position+1,
+                                                          field_position+n_components,
+                                                          field_name));
+              }
+          }
+
+
+        // Third build the actual patch data
+        patches.resize(particles.size());
+
+        typename std::multimap<aspect::Particle::types::LevelInd, aspect::Particle::Particle<dim> >::const_iterator particle = particles.begin();
+
+        for (unsigned int i=0; particle != particles.end(); ++particle, ++i)
+          {
+            patches[i].vertices[0] = particle->second.get_location();
+            patches[i].patch_index = i;
+            patches[i].n_subdivisions = 1;
+            patches[i].data.reinit(property_information.n_components()+1,1);
+
+            patches[i].data(0,0) = particle->second.get_id();
```

Comment:
I think it might be worthwhile specializing `Patch<0,spacedim>` to avoid some of the member variables. But that can wait.